### PR TITLE
Fixes to Email Templates

### PR DESCRIPTION
**Pull request by benzap**
- Fixes Issue #233
- Added the argument `--nostart` to dev.cljs to prevent mounting while in development mode
  - This was so I didn't have to deal with mounting everything, while working on the emailer. Let me know if you want me to remove it.
- Added hiccups library
  - Used with the templates
- Split up emailer functions for easier testing
  - These are tied into the original emailer. (This isn't fully tested, need some help with this @fbielejec @jpmonettas)
- Template on SendGrid was rewritten, going to need some guidance on any additional changes.
   - The changes fixed several issues, I haven't noticed any new issues.
- Template is now a bit more responsive, should work on mobiles >480px in width.
- DANK amounts supplied are now rounded to the 3rd decimal place

# Notes
- My testing required that I use **district@district0x.io** as the `:from` field. Additional setup needs to be introduced on SendGrid, maybe @fbielejec has an idea. Once it's setup, it just requires a change in the config.edn file on qa/production
  - [Related Google Support Answer](https://support.google.com/mail/answer/2451690)

# Previews

## Challenge Created Email
```
(in-ns 'memefactory.server.emailer)
(send-challenge-created-email-handler
 {:from "district@district0x.io"
  :to "ben@district0x.io"
  :title "Cat Meme 8"
  :meme-url "google.com"
  :button-url "duckduckgo.org"
  :meme-image-url "https://i.imgur.com/40bQwiB.jpg"
  :time-remaining "1 Hour Left"
  :on-success #(.log js/console "Success!")
  :on-error #(.error js/console "Error!" %)
  :template-id "REDACTED"
  :api-key "REDACTED"
  :print-mode? false})
```

## Auction Bought Email
```
(in-ns 'memefactory.server.emailer)
(send-auction-bought-email-handler
 {:from "district@district0x.io"
  :to "ben@district0x.io"
  :title "Cat Meme 8"
  :meme-url "google.com"
  :button-url "duckduckgo.org"
  :meme-image-url "https://i.imgur.com/40bQwiB.jpg"
  :on-success #(.log js/console "Success!")
  :on-error #(.error js/console "Error!" %)
  :template-id "REDACTED"
  :api-key "REDACTED"
  :print-mode? false})
```

## Vote Reward Claimed
```
(in-ns 'memefactory.server.emailer)
(send-vote-reward-claimed-email-handler
 {:from "district@district0x.io"
  :to "ben@district0x.io"
  :title "Cat Meme 8"
  :option 1
  :amount (web3/to-wei 2.33333333 :ether)
  :meme-url "google.com"
  :button-url "duckduckgo.org"
  :meme-image-url "https://i.imgur.com/40bQwiB.jpg"
  :on-success #(.log js/console "Success!")
  :on-error #(.error js/console "Error!" %)
  :template-id "REDACTED"
  :api-key "REDACTED"
  :print-mode? false})
```

## Challenge Reward Claimed
```
(in-ns 'memefactory.server.emailer)
(send-challenge-reward-claimed-email-handler
 {:from "district@district0x.io"
  :to "ben@district0x.io"
  :title "Cat Meme 8"
  :amount (web3/to-wei 2.33333333 :ether)
  :meme-url "google.com"
  :button-url "duckduckgo.org"
  :meme-image-url "https://i.imgur.com/40bQwiB.jpg"
  :on-success #(.log js/console "Success!")
  :on-error #(.error js/console "Error!" %)
  :template-id "REDACTED"
  :api-key "REDACTED"
  :print-mode? false})
```

# Possible Changes
- Include more information in the subject text
- Permanent margins around hero for smaller widths?

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -62,6 +62,7 @@
                  [district0x/re-frame-ipfs-fx "0.0.2"]
                  [funcool/bide "1.6.1-SNAPSHOT"] ;; version with fix for duplicated query params
                  [garden "1.3.5"]
+                 [hiccups "0.3.0"]
```
Comment (madvas) [inline project.clj]: remove

**Comment by madvas** [comment]
Please change "Hour Left" to "hour left"
image

**Comment by madvas** [comment]
Change auction sold email text accordingly: https://github.com/district0x/memefactory/issues/233#issuecomment-478487748

**Comment by benzap** [comment]
- Removed Hiccups
- Changed Auction Template String to resemble what was seen in the issue

# Preview

**Comment by benzap** [comment]
@madvas should the address be a link to the memefolio?

**Comment by madvas** [comment]
Cool thanks, just the address should be underlined, because it should be link leading to user's Memefolio - Collected

**Comment by madvas** [comment]
Actually, there shouldn't be "by [address]" but "to [address]", my bad

**Comment by benzap** [comment]
@madvas, pushed your other change as well

**Comment by madvas** [comment]
Thanks!

**Pull request merged**